[GStreamer][GL] Enable DMABuf sink

philn opened the pull request:
#### e962336d3261fc58317d3a0e90032e5c900a79e4
<pre>
[GStreamer][GL] Enable DMABuf sink
<a href="https://bugs.webkit.org/show_bug.cgi?id=252826">https://bugs.webkit.org/show_bug.cgi?id=252826</a>

Reviewed by Žan Doberšek and Xabier Rodriguez-Calvar.

Unconditionally enable the DMABuf sink when running with GStreamer 1.22 and newer. The
WEBKIT_GST_DMABUF_SINK_DISABLED environment variable can be set to fallback to the GL sink.

* Source/WebCore/platform/graphics/gstreamer/DMABufVideoSinkGStreamer.cpp:
(webKitDMABufVideoSinkGetProperty):
(webKitDMABufVideoSinkIsEnabled):
(webKitDMABufVideoSinkProbePlatform):
* Tools/Scripts/webkitpy/port/glib.py:
(GLibPort.setup_environ_for_server):

Canonical link: <a href="https://commits.webkit.org/260881@main">https://commits.webkit.org/260881@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bd1b6f228121330cf146d28cdbdbbd2667074dba

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/109605 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/18729 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/42357 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/1093 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/118736 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/113503 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/20195 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/9926 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/101876 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/115361 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/15032 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/98266 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/43247 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/113074 "Passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/97006 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/29919 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/85026 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/11464 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/31261 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/12122 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/8193 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/17487 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/50870 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7548 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/13862 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

zdobersek commented on Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp:
```diff
@@ -189,7 +189,7 @@ MediaPlayerPrivateGStreamer::MediaPlayerPrivateGStreamer(MediaPlayer* player)
         [&]() -> Ref<TextureMapperPlatformLayerProxy> {
 #if USE(TEXTURE_MAPPER_DMABUF)
-            if (webKitDMABufVideoSinkIsEnabled() && webKitDMABufVideoSinkProbePlatform())
+            if (webKitDMABufVideoSinkProbePlatform())
```
These two conditions should match conditions in `createVideoSinkDMABuf()`. Otherwise things can get weird with using TextureMapperPlatformLayerProxyDMABuf along with a GL sink.

So I would recommend keeping `webKitDMABufVideoSinkIsEnabled()`, and using `WEBKIT_GST_DISABLE_DMABUF_SINK` env to disable it. That way behavior between DMABuf sink and GL sink can be examined by selectively disabling one or the other, as long as they coexist.

philn replied:
Makes sense! I'll update the PR, thanks for the review.

philn replied:
Maybe we can make this work with `GST_PLUGIN_FEATURE_RANK=webkitdmabufsink:0` too. I can try that, I've been wondering lately about avoid new env vars when we can...

philn replied:
GST_PLUGIN_FEATURE_RANK works only during `gst_init_check()` so for our custom elements registered afterwards we'd need to re-parse the env var... Not sure it's worth.

calvaris reviewed (approved):
Other than what @zdobersek said, I think we would be good to go.

philn committed:
[GStreamer][GL] Enable DMABuf sink
https://bugs.webkit.org/show_bug.cgi?id=252826

Reviewed by Žan Doberšek and Xabier Rodriguez-Calvar.

Unconditionally enable the DMABuf sink when running with GStreamer 1.22 and newer. The
WEBKIT_GST_DMABUF_SINK_DISABLED environment variable can be set to fallback to the GL sink.

* Source/WebCore/platform/graphics/gstreamer/DMABufVideoSinkGStreamer.cpp:
(webKitDMABufVideoSinkGetProperty):
(webKitDMABufVideoSinkIsEnabled):
(webKitDMABufVideoSinkProbePlatform):
* Tools/Scripts/webkitpy/port/glib.py:
(GLibPort.setup_environ_for_server):

Canonical link: https://commits.webkit.org/260881@main
